### PR TITLE
chore: use Jest's own configuration types

### PR DESCRIPTION
**Pull request by SimenB**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/jest/blob/895136f3f4be9155f1b60b55e77b8c285feb6a73/packages/jest-types/src/Config.ts
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

This has the same error as #41838, help appreciated 🙂

### CODE REVIEW

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -1553,257 +1554,6 @@ declare namespace jest {
...
-        providesModuleNodeModules: string[];
-    }
-
-    type ReporterConfig = [string, object];
-
-    type ConfigGlobals = object;
-
-    type SnapshotUpdateState = 'all' | 'new' | 'none';
-
-    interface DefaultOptions {
```
Comment (SimenB) [inline types/jest/index.d.ts]: I'm not sure why these are even declared here?

**Review by wsmd — APPROVED** [review]
Thanks for this! Exciting to see `@jest/types` used here! CI is failing for some reason.. can't figure out why, I see that's it's failing on #41838 too 😕

File: types/jest/index.d.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
+/// <reference types="@jest/types" />
```
[the remark this answers is not in the document]
Comment (SimenB) [reply types/jest/index.d.ts]: There was (is) a bug in the publisher which requires this, see #32891

**Comment by SimenB** [comment]
Let's do #41852 instead

**Pull request closed without merging**